### PR TITLE
Fix the slow execution of the CfChainsSimplification pass

### PR DESCRIPTION
**Pull request by hkapp**
This PR fixes the execution speed issues with the CfChainsSimplification pass. This is done by computing the UseDef only once per method, instead of once per Jump instruction encountered (!)

### CODE REVIEW

File: tools/src/main/scala/scala/scalanative/optimizer/pass/CfChainsSimplification.scala [hunk 1]
```diff
@@ -14,20 +14,22 @@ class CfChainsSimplification(implicit top: Top) extends Pass {
   import CfChainsSimplification._
 
   override def onInsts(insts: Seq[Inst]): Seq[Inst] = {
-    val cfg = ControlFlow.Graph(insts)
+    val cfg    = ControlFlow.Graph(insts)
+    val usedef = UseDef(cfg)
+    val method = MethodInfo(cfg, usedef)
```
Comment (densh) [inline tools/src/main/scala/scala/scalanative/optimizer/pass/CfChainsSimplification.scala]: The code would be a bit more clear if you make `MethodInfo` an implicit argument. This way you don't have to manually pass it around in recursive calls down the chain.

Comment (hkapp) [reply tools/src/main/scala/scala/scalanative/optimizer/pass/CfChainsSimplification.scala]: I changed it, is that what you had in mind ?

Comment (densh) [reply tools/src/main/scala/scala/scalanative/optimizer/pass/CfChainsSimplification.scala]: Yep, it's much better now! 👍

**Comment by densh** [comment]
LGTM, thanks for the fix @hkapp ! 👍

**Pull request merged**